Lib/ldap/types.py

```diff
@@ -0,0 +1,58 @@
+"""
+types - type annotations which are shared across modules
+
+See https://www.python-ldap.org/ for details.
+"""
+from ldap.pkginfo import __version__
+
+from typing import (
+    List,
+    MutableMapping,
+    Tuple,
+    Sequence,
+    Optional,
+)
+from typing_extensions import TypeAlias
+
+__all__ = [
+    'LDAPModListAddEntry',
+    'LDAPModListModifyEntry',
+    'LDAPModListEntry',
+    'LDAPAddModList',
+    'LDAPModifyModList',
+    'LDAPModList',
+    'LDAPEntryDict',
+    'LDAPControl',
+    'LDAPControls',
+    'LDAPSearchResult',
+]
+
+LDAPModListAddEntry: TypeAlias = "Tuple[str, List[bytes]]"
+"""The type of an addition entry in a modlist."""
+
+LDAPModListModifyEntry: TypeAlias = "Tuple[int, str, Optional[List[bytes]]]"
```

Comment:
`modify_s()` also supports bytes in addition to `list[bytes] | None`

```suggestion
LDAPModListModifyEntry: TypeAlias = "Tuple[int, str, Union[List[bytes], bytes, None]]"
```

https://www.python-ldap.org/en/python-ldap-3.3.0/reference/ldap.html#ldap.LDAPObject.modify

> mod_vals is either a string value or a list of string values to add, delete or replace respectively.

Reply:
Also in practice it accepts `int`, my code has been using that and it works.

But it's not documented. Should we err on the side of strictness and not allow `int` here?

Reply:
I think the type hints can be stricter than what the code actually accepts...they're supposed to document the "right" way of doing things after all.

Reply:
And good catch WRT `mod_vals`...I also noticed that the `timeout` parameter type (in various functions in `Lib/ldap/ldapobject.py`) is too strict (it accepts `int | float`)